keps/sig-api-machinery/2885-server-side-unknown-field-validation/README.md

```diff
@@ -491,6 +476,34 @@ TypedValue](https://github.com/kubernetes/kubernetes/blob/9ff3b7e744b34c099c1405
 object has any unknown or duplicate fields.
 
 
+#### Kubectl Flag
+
+For beta, we will modify the kubectl `--validate` flag from being a bool flag to
+a string flag that accepts the following values:
+
+* `server`: This sends a request with the fieldValidation param set to `Strict`
+for strict server-side validation.
+* `client`: This performs client-side validation in kubectl that breaks if it
+encounters any validation errors before sending a request to the server. If no
+client-side validation errors are found, the request sent to the server has
+fieldValidation param set to `Ignore`. This will be the default behavior in beta
+as it is identical to the current validation behavior pre-1.24
+* `ignore`: This performs no client-side validation or server-side validation,
+sending a request to the server with fieldValidation param set to `Ignore`.
+* `warn`: This sends a request to the server with the fieldValidation param set
+  to `Warn`. It performs server-side validation, but validation errors are
+  exposed as warnings in the result header rather than failing the request. This
+  will be the default once server-side validation goes GA.
+* `strict`: same behavior as `--validate=server`.
```

Comment:
Why do we need it then?

Reply:
"Strict", "Warn", and "Ignore" are the semantics used by the fieldValidation query param and will be permanent.

"Server" and "Client" are transitionary semantics for now since we want to give the caller the ability to choose until client side validation is no longer an option. It makes sense to offer both "server" and "strict" for now so that the user sees the expected behavior regardless of which set of semantics they are using.